lk2nd/device/dts/msm8952/msm8937-huawei-aum.dts

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <skeleton64.dtsi>
+#include <lk2nd.dtsi>
+
+/*
+ * To flash lk2nd onto the device, you have to manually splice the
+ * resulting image into two separate images: ramdisk and kernel.
+ */
```

Comment:
I'd appreciate if you could give a bit more detailed instructions here (i.e. how do you generate the kernel image, I think you can ignore the ramdisk part since lk2nd doesn't use it anyway)

Reply:
Not needing to deal with the ramdisk simplifies it a bit, then.

Reply:
Let's see... Would that be good enough?
```
/*
 * To flash lk2nd onto the device, you need to flash lk2nd as the
 * kernel and flash an empty image in place of the ramdisk.
 * To create an empty image use the following command:
 * mkbootimg --kernel /dev/null --ramdisk /dev/null --base 0x80000000 --kernel_offset 0x00008000 --ramdisk_offset 0x02000000 --second_offset 0x00f00000 --tags_offset 0x00000100 --pagesize 2048 --header_version 0 -o ramdisk.img"
 */
```

Reply:
The only thing left is to create the boot partition... The only way to boot anything right now is using <b>fastboot boot</b>.

Reply:
And as for the partition:
```
/*
 * As this device has no boot partition, you'll have to create
 * one yourself.
 * To create the partition follow these steps:
 * ./parted /dev/block/mmcblk0
 * unit kB
 * resizepart 55 31209799kB
 * mkpart boot ext2 31209799kB 100%
 * quit
 * dd if=/dev/block/mmcblk0p55 of=/dev/block/mmcblk0p56 bs=1 seek=446 count=64
 */
```

Reply:
fwiw you don't need boot.img at all if your only goal is to boot mainline :)

Reply:
Via extlinux.conf, I know. But, imo, the existence of that boot partition simplifies quite a few things with this phone.